replica/table.cc

```diff
@@ -549,8 +549,17 @@ void table::enable_off_strategy_trigger() {
 
 storage_group_manager::~storage_group_manager() = default;
 
+storage_group* storage_group_manager::storage_group_for_id(const schema_ptr& s, size_t i) const {
```

Comment:
I think we can just modify `storage_group_of` to return `std::pair<storage_group*, locator::tablet_range_side>`